spec/context.md

```diff
@@ -56,10 +56,39 @@ The shared dictionary is optional.
 LinkToSharedDictionary ::= TBD
 ```
 
+# Chunk
+
+A file may contain one or more **Chunks**. Typically, the first **Chunk** will contain the code
+known to be executed during the startup, additional chunks may contain the definition of lazy
+functions. Additionally, if several source files are concatenated into one, there may be several
+**Chunks**, each with its own eager code.
+
+```
+Chunk  ::= ChunkHeader Prelude? Content? Main Footers?
+ChunkHeader ::= "CHNK" ByteLen ChunkIdentifier
+ChunkIdentifier ::= "EAGR" ChunkNum
+                 |  "LAZY" ChunkNum LazyNum
+ChunkNum ::= var_u32
+LazyNum ::= var_u32
+```
+
+A `ChunkIdentifier` may be:
+
+- `EAGR`, in which case the chunk contains toplevel code, meant to be parsed and executed immediately;
+- `LAZY`, in which case the chunk contains lazy function definitions, meant to be parsed/executed by-need.
+
+It is a syntax error for an `EAGR` chunk to appear after a `LAZY` chunk.
```

Comment:
So, when concatenating multiple files both with eager and lazy chunks, they should be sorted, right?

Reply:
Yes. I figure that it should be fairly easy to do. If the `ChunkNum` do not collide, it's just taking 4 ranges of bytes and reordering them.

Note: Maybe `ChunkNum` should be a fixed length number, rather than a `var_u32`.